flaml/fabric/mlflow.py

```diff
@@ -354,6 +354,7 @@ def _pickle_and_log_artifact(self, obj, artifact_name, pickle_fpath="temp_.pkl")
         with open(pickle_fpath, "wb") as f:
             pickle.dump(obj, f)
         mlflow.log_artifact(pickle_fpath, artifact_name)
+        os.remove(pickle_fpath)
```

Comment:
what if `log_artifact` fails?

Reply:
the code would look like

```python
with tempfile.TemporaryDirectory() as tmpdir:
    pickle_path = tmpdir/path
    # save pickle
    mlflow.log_artifact(pickle_path)

# pickle file deleted when exiting the context manager even if log_artifact fails
```

Reply:
Creating files in the current working directory is not preferred. Imagine you're working in a code editor, and a pickle file suddenly shows up and disappears.

Reply:
@harupy,  thanks for the suggestions, updated.